plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraTypeManager.java

```diff
@@ -308,11 +316,42 @@ private String buildArrayValueFromSetType(GettableByIndex row, int position, Dat
         return buildArrayValue((Collection<?>) row.getObject(position), setType.getElementType());
     }
 
-    private String buildArrayValueFromListType(GettableByIndex row, int position, DataType type)
+    private Block buildArrayValueFromListType(GettableByIndex row, int position, DataType type)
     {
         checkArgument(type instanceof ListType, "Expected to deal with an instance of %s class, got: %s", ListType.class, type);
         ListType listType = (ListType) type;
-        return buildArrayValue((Collection<?>) row.getObject(position), listType.getElementType());
+        ArrayType trinoArrayType = getTrinoArrayType(listType.getElementType());
+        return buildArrayValueFromCollection((Collection<?>) row.getObject(position), trinoArrayType);
+    }
+
+    private Block buildArrayValueFromCollection(Collection<?> collection, ArrayType trinoArrayType)
+    {
+        ArrayBlockBuilder blockBuilder = (ArrayBlockBuilder) trinoArrayType.createBlockBuilder(null, 1);
+        SingleArrayBlockWriter singleArrayBlockWriter = blockBuilder.beginBlockEntry();
+        for (Object value : collection) {
+            value = convertArrayValueAccordingToTrinoType(trinoArrayType.getElementType(), value);
+            writeNativeValue(trinoArrayType.getElementType(), singleArrayBlockWriter, value);
+        }
+        blockBuilder.closeEntry();
+        return trinoArrayType.getObject(blockBuilder, 0);
+    }
+
+    private Object convertArrayValueAccordingToTrinoType(Type elementType, Object value)
+    {
+        if (elementType.getJavaType() == long.class) {
+            if (value instanceof Long) {
```

Comment:
I suggest you support other data types as well. Otherwise, the implementation will be handling limited use cases.